lib/jsonapi/include_directive.rb

```diff
@@ -40,6 +42,28 @@ def [](key)
       end
     end
 
+    # @param another_directive [IncludeDirective]
+    # @return [IncludeDirective]
+    def merge(other)
+      fail ArgumentError,
+           "the value of 'other' MUST be an IncludeDirective" unless
+          other.is_a?(IncludeDirective)
+
+      hash = to_hash.dup
+      Parser.deep_merge!(hash, other.to_hash)
+
+      IncludeDirective.new(hash, options.merge!(other.options))
```

Comment:
Ok I see your point. I took this decision thinking about `allow_wildcard` (which looks to me like the only available option at the moment). But thinking again about it, it definitely doesn't work well.

Currently you only need to specify `allow_wildcard` if you have wildcards in your include.
If this is not the case for `other`, than whatever is in `self` remain. If you need it in your new hash, instead, that will be merged properly in the result.

The only case this is not working as everyone would expect, is when `other.options` explicitly contains `allow_wildcard: false`.